[Jericho Chua] iP

jerichochua opened the pull request.

jerichochua committed:
Add greeting

jerichochua committed:
Add echo command

jerichochua committed:
Add ability to store and list user input

jerichochua committed:
Add Task class

Add new Task class, as well as modifying the code in Duke.java
to make use of the new class.

jerichochua committed:
Add ability to mark tasks as done

jerichochua committed:
Add Todo, Deadline and Event classes

jerichochua committed:
Override toString method in Task class

jerichochua committed:
Add support for tracking todos, deadlines and events

* Rearrange code for clarity
* Remove validation (for now)
* Create new method to add user tasks

jerichochua committed:
Add variable to print 'task' and 'tasks'

jerichochua committed:
Add semi-automated text UI testing

jerichochua committed:
Comply with coding standard

jerichochua committed:
Extract userTasks as a class-level attribute

hungvo0603 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,65 @@
+
+        userInput = in.nextLine();
+
+        while (!userInput.equals("bye")) {
+            if (userInput.equals("list")) {
+                listUserTasks(userTasksCount);
+            } else if (userInput.contains("done")) {
+                int taskNumber = Integer.parseInt(userInput.substring(5));
+                markTaskAsDone(taskNumber);
+            } else {
+                addUserTask(userInput);
+            }
+            userInput = in.nextLine();
+        }
```
Should command words like "bye", "list", "done" be defined in an enumeration?

hungvo0603 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,65 @@
+    }
+
+    public static void addUserTask(String task) {
+        if (task.startsWith("todo")) {
+            task = task.replace("todo", " ").trim();
+            userTasks[userTasksCount] = new Todo(task);
+        } else if (task.startsWith("deadline")) {
+            task = task.replace("deadline", " ").trim();
+            String[] taskSplit = task.split(" /by ");
+            userTasks[userTasksCount] = new Deadline(taskSplit[0], taskSplit[1]);
+        } else if (task.startsWith("event")) {
+            task = task.replace("event", " ").trim();
+            String[] taskSplit = task.split(" /at ");
+            userTasks[userTasksCount] = new Event(taskSplit[0], taskSplit[1]);
+        } else {
+            return;
+        }
```
Should command words like "todo", "deadline", "event" be defined in an enumeration?

hungvo0603 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,13 @@
+public class Event extends Task {
+    protected String eventAt;
```
Should this variable mention something about time? eventAt may have two meanings 1) location of the event 2) time of the event.

syncode98 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,65 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String userInput;
+        Scanner in = new Scanner(System.in);
+
+        System.out.println("Hello! I'm Duke" + System.lineSeparator() + "What can I do for you?");
+
+        userInput = in.nextLine();
```
Hi Jericho. Your code is readable and there are no other serious issues. However, I would like to suggest that you do not need to have the spaces in lines 11 and 13. I think that your code would still be readable even without these spaces. However, if you do feel that these spaces improves readability, do implement them.

jerichochua committed:
Declare Task class as abstract

jerichochua committed:
Implement basic error handling

jerichochua committed:
Update test case

jerichochua committed:
Add tab characters for output

jerichochua committed:
Merge branch 'branch-Level-5' into master

jerichochua committed:
Organise classes into duke package

jerichochua committed:
Organise some classes into task package

jerichochua committed:
Merge branch 'branch-A-Packages' into master

syncode98 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,15 @@
+package duke.task;
+
+public class Deadline extends Task {
+    protected String by;
+
```
Consider renaming the String to something other than by, like deadlineBy etc.

syncode98 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
+    protected String by;
+
```
Perhaps consider changing the name of the String variable to deadline or dueDate.

syncode98 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,30 @@
+
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718");
+    }
```
Perhaps you could provide a comment regarding what these two Unicode represent. You can define them separately if you want to. This would make it easier to understand what they are.

jerichochua committed:
Add exceptions

jerichochua committed:
Update test case

jerichochua committed:
Merge branch 'A-Exceptions' into master

jerichochua committed:
Replace magic literals with constants

jerichochua committed:
Process user input in a new method

jerichochua committed:
Make constants private

jerichochua committed:
Refactor code under processUserInput method

Refactor code under processUserInput method, so that new commands can
be added easily. Also abstracted the adding of tasks into separate
methods.

jerichochua committed:
Catch exception when invalid task number is entered

jerichochua committed:
Abstract greeting messages to separate methods

jerichochua committed:
Replace userTasks array with ArrayList

jerichochua committed:
Add support for deleting tasks

jerichochua committed:
Catch exception when marking task as done with invalid number

jerichochua committed:
Add support for saving tasks to a file

jerichochua committed:
Add support for reading tasks from a file

jerichochua committed:
Handle case when task is already marked as done in the file

jerichochua committed:
Merge branch 'branch-Level-6' into master

jerichochua committed:
Merge branch 'branch-Level-7' into master

jerichochua committed:
Update tests

jerichochua committed:
Update tests to include more cases

jerichochua committed:
Replace tick and cross in Task class to use letters instead

jerichochua committed:
Initial refactoring of code by extracting classes

jerichochua committed:
Refactor and extract code to create new Storage class

jerichochua committed:
Create new Parser class and refactor code

jerichochua committed:
Refactor code

jerichochua committed:
Create new Command class and its subclasses

jerichochua committed:
Add isExit abstract method in Command class

Implemented the isExit method in its subclasses, and made use of
isExit method in the Duke class as well.

jerichochua committed:
Add new parameters to execute method of Command class

jerichochua committed:
Refactor code

jerichochua committed:
Refactor more code

jerichochua committed:
Fix exception handling

Mick609 commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,91 @@
+    }
+
+    public Task addDeadline(String[] arguments) throws EmptyDescriptionException {
+        if (arguments.length != 2) {
```
If the argument length is not 2, it does not necessarily mean the description is empty. What if there are more than 2 arguments?

jerichochua committed:
Refactor Parser, Storage and TaskList classes

Mick609 commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,34 @@
+
+public class DeadlineCommand extends Command {
+    public static final String COMMAND_DEADLINE = "deadline";
+    private final String arguments;
```
Maybe the name of a single string can be "argementsString".

jerichochua committed:
Refactor Duke class

Mick609 commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,31 @@
+    public static final String COMMAND_DONE = "done";
+    private final int taskNumber;
+
+    public DoneCommand(String arguments) {
```
Maybe you can consider changing the name of the string variable to something like "argumentString" or "argumentStr".

Mick609 commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,34 @@
+    public static final String COMMAND_EVENT = "event";
+    private final String arguments;
+
+    public EventCommand(String arguments) {
```
A similar suggestion here. Change the name of the variable to a singleton.

Mick609 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,19 @@
+package duke.task;
+
+public class Deadline extends Task {
+    protected String by;
```
How about changing the name of this variable to "deadlineTime"?

Mick609 reviewed (commented):
Wonderful job. Easy to read with minimum problems.

jerichochua committed:
Resolve issues in code based on comments

jerichochua committed:
Add support for finding tasks

jerichochua committed:
Explicitly list imported classes in Parser class

jerichochua committed:
Add Javadoc comments to Duke and Command classes

jerichochua committed:
Add Javadoc comments to TaskList class

jerichochua committed:
Add Javadoc comments to Ui class

jerichochua committed:
Add more Javadoc comments

jerichochua committed:
Add Javadoc comments to exception classes

jerichochua committed:
Merge pull request #1 from jerichochua/branch-Level-9

Merge Level-9 branch

jerichochua committed:
Merge branch 'master' into branch-Level-9

jerichochua committed:
Merge branch 'master' into branch-A-JavaDoc

jerichochua committed:
Merge pull request #2 from jerichochua/branch-A-JavaDoc

Merge A-JavaDoc branch

jerichochua committed:
Merge branch 'master' into branch-A-JavaDoc

jerichochua committed:
Add support for recognising dates in deadline and event tasks

jerichochua committed:
Add support for recognising time in deadline and event tasks

jerichochua committed:
Catch exception when incorrect date/time is entered

jerichochua committed:
Merge pull request #3 from jerichochua/branch-Level-8

Merge Level-8 branch

jerichochua committed:
Update tests

jerichochua committed:
Update user guide

jerichochua committed:
Set theme jekyll-theme-cayman

jerichochua committed:
Minor changes

* Added some Javadoc comments
* Minor refactoring

jerichochua committed:
Add a new exception class

Added a new exception class to deal with invalid index entry,
when marking a task as done or deleting a task.

jerichochua committed:
Update expected output for testing

jerichochua committed:
Merge branch 'branch-Level-9' into master

jerichochua committed:
Merge branch 'branch-A-JavaDoc' into master

jerichochua committed:
Add clear command

jerichochua committed:
Update user guide to include clear command

jerichochua committed:
Fix issue when user enters a vertical bar character

When a user enters a vertical bar character in the task description, and it is saved in the file, when the file is opened again, it causes issues with the parsing of the file. Now, the user is no longer allowed to use the vertical bar character for the task description. User guide is updated to reflect this change.

jerichochua committed:
Update user guide

jerichochua committed:
Refactor code

* Trimmed whitespaces when arguments are supplied
* Use regex for parsing commands and arguments
* Other minor changes

Should not have any regressions.

jerichochua committed:
Update user guide

jerichochua committed:
Minor changes
